bank2ynab.py

```diff
@@ -622,14 +646,33 @@ def _fix_decimal_point(self, row):
 
         return row
 
+    def _clean_monetary_values(self, row):
+        """
+        remove any characters from inflow or outflow strings except
+        digits and '.'
+        :param row: list of values
+        """
+        inflow_index = self.config["output_columns"].index("Inflow")
+        outflow_index = self.config["output_columns"].index("Outflow")
+        row[inflow_index] = re.sub(r"[^\d\.]", "", row[inflow_index])
+        row[outflow_index] = re.sub(r"[^\d\.]", "", row[outflow_index])
+
+        return row
+
     def _valid_row(self, row):
-        """ if our row doesn't have an inflow or outflow, mark as invalid
+        """ if our row doesn't have an inflow, outflow or a valid date,
+        mark as invalid
         :param row: list of values
         """
         inflow_index = self.config["output_columns"].index("Inflow")
         outflow_index = self.config["output_columns"].index("Outflow")
         if row[inflow_index] == "" and row[outflow_index] == "":
             return False
+        # check that date matches YYYY-MM-DD format
+        if self.config["discard_invalid_dates"]:
```

Comment:
I think we can remove `if self.config["discard_invalid_dates"]:` and the relevant config param- we should never allow a row with an invalid date. We should include an error message telling anybody with a failing bank the link to go to with info on date formats.

Reply:
OK. I'm pulling it out, but I am now going to need to fix a few test cases that are failing because the dates aren't in the right format. Will try to do that in a bit.